### PR TITLE
860 activity task ssantiago

### PR DESCRIPTION
**Pull request by DJSanti**
Opening up for edits and suggestions while I switch to something else for a little while.

### CODE REVIEW

File: docs/concepts/what-is-an-activity-task.md [hunk 1]
```diff
@@ -2,9 +2,27 @@
...
-description: An Activity Task is a Task that contains the context needed to make progress with an Activity Execution.
+description: An Activity Task contains the context needed to make an Activity Task Execution.
 tags:
   - explanation
 ---
 
-An Activity Task is a Task that contains the context needed to make progress with an Activity Execution.
+An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).
+
+Activity Tasks are meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once. Should this fail, the Activity Task is retried until it's completed successfully.
```
[the remark this answers is not in the document]
Comment (djmagee) [reply docs/concepts/what-is-an-activity-task.md:12]:
```suggestion
Activity Tasks are meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once.
If an Activity Task Execution fails, the Activity Task is retried in accordance with its Retry Policy.
```

File: docs/concepts/what-is-an-activity-task.md [hunk 2]
```diff
@@ -2,9 +2,27 @@
...
+
+#### Scheduling Activity Tasks
+
+Activity Tasks are assigned to an Activity Task Queue upon an [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks). Once a Task is executed, an Activity's Events are run until the Activity successfully completes.
```
[the remark this answers is not in the document]
Comment (djmagee) [reply docs/concepts/what-is-an-activity-task.md:16]:
```suggestion
Activity Tasks are assigned to an Activity Task Queue upon an [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks).
```
An Activity doesn't run Events. Events are generated by the Temporal Server.

File: docs/concepts/what-is-an-activity-task.md [hunk 3]
```diff
@@ -2,9 +2,27 @@
...
+
+#### Context Events
+
+Activity Tasks contain Events that contain what's needed to execute the Activity. These events are found within the Activity Task's [Context](/docs/concepts/what-is-an-activity-task#context-events).
```
[the remark this answers is not in the document]
Comment (djmagee) [reply docs/concepts/what-is-an-activity-task.md:20]: An Activity Task can contain context, but Events are generated by the Temporal Server.

File: docs/concepts/what-is-an-activity-task.md [hunk 4]
```diff
@@ -2,9 +2,27 @@
...
+
+[event explanation?]
+
+ActivityTaskScheduled prepares the Activity for execution. ActivityTaskStarted begins the Activity.
```
[the remark this answers is not in the document]
Comment (djmagee) [reply docs/concepts/what-is-an-activity-task.md:24]: These Events are generated by the Temporal Server. I'm hesitant to cast them as actors.

File: docs/concepts/what-is-an-activity-task.md [hunk 5]
```diff
@@ -2,9 +2,28 @@
...
 
-An Activity Task is a Task that contains the context needed to make progress with an Activity Execution.
+An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).
+
+Activity Tasks are meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once.
+If an Activity Task Execution fails, the Activity Task is retried in accordance with its Retry Policy.
+
+#### Scheduling Activity Tasks
+
+Activity Tasks are assigned to an Activity Task Queue upon an [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks).
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:17]: I think its more of a correlation -
An Activity Task comes into existence when it is scheduled - and the Event is added to the Workflow's Event History.

File: docs/concepts/what-is-an-activity-task-execution.md [hunk 6]
```diff
@@ -7,6 +7,10 @@ tags:
   - explanation
 ---
 
-An Activity Task Execution is the execution of an Activity function.
+An Activity Task Execution results from the invocation of Activity Function Definition.
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task-execution.md:10]: I wouldn't say it "results from".
An Activity Task Execution is when the Worker uses the context provided from the Activity Task and executes the Activity Function Definition.

Another way to look at it - is it is the period between the ActivityTaskStarted and ActivityTaskCompleted (or other Closed) Events.

File: docs/concepts/what-is-an-activity-task.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (flossypurse) [reply docs/concepts/what-is-an-activity-task.md:25]: Ya, these Events are correlations to other things happening - 

The Temporal Cluster puts an Activity Task into the Task Queue and that correlates to the ActivityTaskScheduled Event.

File: docs/concepts/what-is-an-activity-task.md [hunk 7]
```diff
@@ -2,9 +2,28 @@
...
+
+#### Context Events
+
+Activity Tasks contain Events that are needed to execute the Activity. These events are found within the Activity Task's [Context](/docs/concepts/what-is-an-activity-task#context-events).
+
+[event explanation?]
+
+ActivityTaskScheduled prepares the Activity for execution. ActivityTaskStarted begins the Activity.
+
+Upon completion, the Activity Task responds to the cluster with an Event status of _success_ or _failure_.
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:27]: The Worker responds with a specific Event - (one of ActivityTaskCompleted, ActivityTaskFailed, ActivityTaskTimedOut, ActivityTaskCanceled, ActivityTaskedTerminated)

File: docs/concepts/what-is-an-activity-task.md [hunk 8]
```diff
@@ -2,9 +2,28 @@
...
+
+[failure events may have retry?]
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:29]: Yes, if the Retry Policy dictates it - and if the Events are either ActivityTaskFailed or ActivityTaskTimedOut for example - then yes, the Cluster would schedule another Activity Task.

File: docs/concepts/what-is-an-activity-task-execution.md [hunk 9]
```diff
@@ -7,6 +7,8 @@ tags:
   - explanation
 ---
 
-An Activity Task Execution is the execution of an Activity function.
+An Activity Task Execution is when the Worker uses the Context provided from the [Activity Task](/docs/concepts/what-is-an-activity-task) and executes the [Activity Definition](/docs/concepts/what-is-an-activity-definition).
 
-Activity Task Executions are retried per a Retry Policy.
+From the API, users see an Activity being executed and assume that it runs once.
+
+In reality, the Activity can be retried per a Retry Policy until it achieves its expected result.
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task-execution.md:14]:
```suggestion
The API to schedule an Activity Execution provides an "effectively once" experience, even though there may be several Activity Task Executions that take place to successfully complete an Activity.
```

File: docs/concepts/what-is-an-activity-task.md [hunk 10]
```diff
@@ -2,9 +2,27 @@
...
+From the API, users see an Activity being executed and assume that it runs once.
+
+An Activity Task is meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once.
+If an Activity Task Execution fails, the Activity Task is retried in accordance with its Retry Policy.
+
+#### Context Events
+
+Activity Tasks contain Events that are needed to execute the Activity. These events are found within the Activity Task's [Context](/docs/concepts/what-is-an-activity-task#context-events).
+
+The Temporal Cluster puts an Activity Task into the Task Queue. This correlates to the [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks) in the Activity Task's Context.
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:20]: Each sentence on new line

File: docs/concepts/what-is-an-activity-task.md [hunk 11]
```diff
@@ -2,9 +2,27 @@
...
-An Activity Task is a Task that contains the context needed to make progress with an Activity Execution.
+An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).
+From the API, users see an Activity being executed and assume that it runs once.
+
+An Activity Task is meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once.
+If an Activity Task Execution fails, the Activity Task is retried in accordance with its Retry Policy.
+
+#### Context Events
+
+Activity Tasks contain Events that are needed to execute the Activity. These events are found within the Activity Task's [Context](/docs/concepts/what-is-an-activity-task#context-events).
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:18]: Hmm... why does this link right back to this page?
Also, an Activity Task does not contain Events - nor is it an Event.
Events live in the Workflow Execution Event History.

My current suggestion would drop the "Context Events" header and this sentence above and just have a solid set of sentences / paragraph of info.

File: docs/concepts/what-is-an-activity-task.md [hunk 12]
```diff
@@ -2,9 +2,27 @@
...
+
+The Temporal Cluster puts an Activity Task into the Task Queue. This correlates to the [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks) in the Activity Task's Context.
+
+Upon completion, the Activity Task responds to the cluster with a specific Event:
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:22]: Move this into the Activity Task Execution page -

But the Worker responds to the Cluster once the Activity Task Execution finishes (with any of the below Events)

File: docs/concepts/what-is-an-activity-task-execution.md (hunk 9 above)
Comment (flossypurse) [reply docs/concepts/what-is-an-activity-task-execution.md:14]: Could add more about the Events associated with an Activity Task Execution - like I mention below.

File: docs/concepts/what-is-an-activity-task.md [hunk 13]
```diff
@@ -2,9 +2,27 @@
...
 tags:
   - explanation
 ---
 
-An Activity Task is a Task that contains the context needed to make progress with an Activity Execution.
+An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).
+From the API, users see an Activity being executed and assume that it runs once.
+
+An Activity Task is meant to run an [Activity Definition](/docs/concepts/what-is-an-activity-definition) once.
+If an Activity Task Execution fails, the Activity Task is retried in accordance with its Retry Policy.
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:14]: I think we can drop this bit of info as it is covered in the Activity Task Execution page.

File: docs/php/workers.md [hunk 14]
```diff
@@ -30,15 +30,15 @@ $worker = $factory->newWorker();
 // Workflows are stateful. So you need a type to create instances.
 $worker->registerWorkflowTypes(App\DemoWorkflow::class);
 
-// Activities are stateless and thread safe. 
+// Activities are stateless and thread safe.
```
Comment (flossypurse) [inline docs/php/workers.md:33]: looks like your editor and prettier don't seem to agree about these spaces...

File: docs/concepts/what-is-an-activity-task-execution.md [hunk 15]
```diff
@@ -7,6 +7,8 @@ tags:
   - explanation
 ---
 
-An Activity Task Execution is the execution of an Activity function.
+An Activity Task Execution is when the Worker uses the Context provided from the [Activity Task](/docs/concepts/what-is-an-activity-task) and executes the [Activity Definition](/docs/concepts/what-is-an-activity-definition).
 
-Activity Task Executions are retried per a Retry Policy.
+The Temporal Cluster puts an Activity Task into the Task Queue. This correlates to the [ActivityTaskScheduled Event](/docs/concepts/what-is-an-activity-task#scheduling-activity-tasks) in the Activity Task's Context.
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task-execution.md:12]:
```suggestion
The [ActivityTaskScheduled Event](/docs/references/events/activitytaskscheduled) corresponds to when the Temporal Cluster puts the Activity Task into the Task Queue.

The [ActivityTaskStarted Event](/docs/references/events/#activitytaskstarted) corresponds to when the Worker picks up the Activity Task from the Task Queue.

Either [ActivityTaskCompleted](/docs/references/events/activitytaskcompleted) or one of the other Closed Activity Task Events corresponds to when the Worker has yielded back to the Temporal Cluster.
```

File: docs/concepts/what-is-an-activity-task.md [hunk 16]
```diff
@@ -2,9 +2,20 @@
...
 sidebar_label: Activity Task
-description: An Activity Task is a Task that contains the context needed to make progress with an Activity Execution.
+description: An Activity Task contains the context needed to make an Activity Task Execution.
 tags:
   - explanation
 ---
 
-An Activity Task is a Task that contains the context needed to make progress with an Activity Execution.
+An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).
+From the API, users see an Activity being executed and assume that it runs once.
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:11]: Not sure this feels like this sentence belongs here...

File: docs/concepts/what-is-an-activity-task.md (hunk 12 above)
Comment (flossypurse) [reply docs/concepts/what-is-an-activity-task.md:15]: Activity Tasks do not respond.
Workers respond. 
And I think Activity Task Execution is a better spot for this - merge with my suggestion above maybe.

File: docs/concepts/what-is-an-activity-task.md [hunk 17]
```diff
@@ -2,9 +2,20 @@
...
+
+Activity Tasks contain Events that are needed to execute the Activity. These events are found within the Activity Task's `Context`.
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:13]: I would change this to - 
Activity Tasks largely represent the Activity Task Scheduled Event which contains the data needed to execute an Activity Function.
If Heartbeat data is being passed, an Activity Task will also contain the latest Heartbeat details for example.

Something like that maybe

File: docs/concepts/what-is-an-activity-task.md [hunk 18]
```diff
@@ -2,9 +2,15 @@
...
+description: An Activity Task contains the context needed to make an Activity Task Execution.
 tags:
   - explanation
 ---
 
-An Activity Task is a Task that contains the context needed to make progress with an Activity Execution.
+An Activity Task contains the context needed to proceed with an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).
+
+Activity Tasks contain Events that are needed to execute the Activity. These events are found within the Activity Task's `Context`.
+
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task.md:13]:
```suggestion

```
Just remove this in favor of the sentences below.

**Review by flossypurse — APPROVED** [review]
You are leaving this in a much better place than when you started.
Well done.

File: docs/concepts/what-is-an-activity-task-execution.md [hunk 19]
```diff
@@ -7,6 +7,20 @@ tags:
   - explanation
 ---
 
-An Activity Task Execution is the execution of an Activity function.
+An Activity Task Execution is when the Worker uses the Context provided from the [Activity Task](/docs/concepts/what-is-an-activity-task) and executes the [Activity Definition](/docs/concepts/what-is-an-activity-definition) (also known as the Activity Function).
 
-Activity Task Executions are retried per a Retry Policy.
+The [ActivityTaskScheduled Event](/docs/references/events/activitytaskscheduled) corresponds to when the Temporal Cluster puts the Activity Task into the Task Queue.
```
Comment (flossypurse) [inline docs/concepts/what-is-an-activity-task-execution.md:12]:
```suggestion
The [ActivityTaskScheduled Event](/docs/references/events#activitytaskscheduled) corresponds to when the Temporal Cluster puts the Activity Task into the Task Queue.
```

I apologize - I gave you bad links which is causing the build to fail.

**Pull request merged**